lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1055,6 +1185,9 @@ static LogicalResult checkUseCount(Operation *op, Value res) {
 
 // Checks if op successors are in appropriate blocks
 static LogicalResult checkSuccessorBlocks(Operation *op, Value res) {
```

Comment:
I'm not sure how we should handle cases like the one involving the buffer. The buffer isn't part of any basic block, thus this check breaks down.

Reply:
We could drop this in favor of the tests.